reposilite-backend/src/main/kotlin/com/reposilite/auth/AuthenticationFacade.kt

```diff
@@ -46,6 +46,7 @@ class AuthenticationFacade(
 
     fun registerAuthenticator(authenticator: Authenticator) {
         this.authenticators.add(authenticator)
+        this.authenticators.sortWith(compareByDescending { it.priority() })
```

Comment:
I think we should move to some sort of a sorted collection, e.g. sorted set, because in this case sorting is only applied when we're adding a new authenticator via this method. It's a bit inconsistent considering that we're also passing some of the authenticators via constructor.